src/nvim/shada.c

```diff
@@ -3413,8 +3413,16 @@ static ShaDaReadResult shada_read_next_item(ShaDaReadDef *const sd_reader,
     return mru_ret;
   }
 
-  const size_t length = (size_t) length_u64;
-  entry->timestamp = (Timestamp) timestamp_u64;
+  if (length_u64 > PTRDIFF_MAX) {
+    emsgf(_(RCERR "Error while reading ShaDa file: "
+            "there is an item at position %" PRIu64 " "
+            "that is stated to be too long"),
```

Comment:
Why "is stated to be" instead of just "is"? Could be abbreviated without ambiguity:

```c
emsgf(_(RCERR "Error while reading ShaDa file: "
        "item at position %" PRIu64 " is too long"),